src/components/Dropdown/DropdownInput.tsx

```diff
@@ -37,15 +37,15 @@ export function DropdownInput(props: {
 
   const handleChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
     toggleDropdown(true);
+    onChange?.(e.target.value);
     updateFocusedItem(-1, e.target.value);
     setLastTypedOrSubmittedValue(e.target.value);
-    onChange?.(e.target.value);
   }, [onChange, setLastTypedOrSubmittedValue, toggleDropdown, updateFocusedItem]);
 
   const handleKeyDown = useCallback((e: KeyboardEvent<HTMLInputElement>) => {
     if (e.key === 'Enter' && (!submitCriteria || submitCriteria(focusedIndex))) {
+      updateFocusedItem(focusedIndex);
```

Comment:
why do we need this change?

Reply:
what happen if user hit enter in the input box when there are no options? Should it clear the search or leave it as is with nothing selected?

Reply:
Ooh yeah it shouldn't trigger a search. I'll need to revert my commit where I removed the `submitCriteria`

Reply:
I think this wouldn't be a problem related to removing `submitCriteria`, because the text wouldn't match any filter value since there are none in the dropdown? but not sure if it should clear the search or leave it as is without selecting anything

Reply:
I reverted it anyways because if there are no filters in the dropdown, the focused index is -1, which means that the check is still needed. So if the user hits enter on an invalid filter nothing would happen